### ISSUE TITLE
feat: Add Dockerfile for container-based deploys

### ISSUE THREAD

**Issue by coopernetes** [finos/git-proxy#275]
**Is your feature request related to a problem? Please describe.**
It's very common in cloud-based deployments to ship a Docker/OCI container for web applications & servers. It would be useful if we provided a default Dockerfile for users building & deploying git-proxy from source to be able to conveniently build the project and create a Docker image. 

As part of a release of git-proxy, the project could also publish a Docker image to [Docker Hub](https://hub.docker.com) under a FINOS namespace for users to simply reuse directly.

**Describe the solution you'd like**
- add a Dockerfile to the repository to build git-proxy as a container image (see https://github.com/finos/git-proxy/pull/269/commits/095cc3a1b48db5ab13fce152585045567282e0e8 for an example)
- build & push an image as part of the release process

**Describe alternatives you've considered**
- we already publish a runnable package to NPM so this may be extraneous. A user Dockerfile could be something as simple as:

```Dockerfile
FROM node:18
ARG GITPROXY_VERSION="1.0.0"

RUN npm install -g @finos/git-proxy@$GITPROXY_VERSION

ENTRYPOINT ["git-proxy"]
```

**Comment by Rajdeep1311** [comment]
I would like to see to this issue.

**Comment by JamieSlome** [comment]
@Rajdeep1311 - awesome, thanks for getting involved! 🤝

I'll assign the issue to you!

**Comment by Rajdeep1311** [comment]
Thanks sir @JamieSlome

**Comment by JamieSlome** [comment]
@Rajdeep1311 - of course! Feel free to post any questions you have here.

I am happy to help and I am sure @coopernetes will be able to provide some guidance as well.

How can we help?

**Comment by Rajdeep1311** [comment]
I created a dockerfile, built a docker image out of it, and ran it but having some issues with the port, it should run on 3000 but firewall problems I guess

**Comment by coopernetes** [comment]
@Rajdeep1311 git-proxy runs on 8000 for the proxy server and 8080 for the management UI. You will need to expose those ports and not 3000.

**Comment by Rajdeep1311** [comment]
Okay sir

**Comment by Rajdeep1311** [comment]
Sir actually I am new to DevOps, Can you help which file is to be used to create the docker file @coopernetes

**Comment by Rajdeep1311** [comment]
Also, I have tried something, if you can see to it, I would like to demonstrate @coopernetes

**Comment by coopernetes** [comment]
> Also, I have tried something, if you can see to it, I would like to demonstrate @coopernetes 

Please submit a pull request with your proposed Dockerfile and I can review it.

**Comment by kyet** [comment]
Hi @JamieSlome! This issue should be re-opened.

In #269, commit 7c2d7ca845872d726ef308600755cfbb771fa7bf added a Dockerfile,
and then commit c5cd565d38dfd0bf8f61b63c01b02923a265c1a9 removed it again.
The reason for this was likely because related work was being done in #275.

Ultimately, #269 was merged without a Dockerfile, and #275 was closed without being merged.
In conclusion, #269 did not fix this issue. Did I miss anything?
